Ernesti/Kirchholtes Random Walker

ArkayHub opened the pull request.

Commit:
implemented

Commit:
Added documentation and configuration

ArkayHub committed:
Update README.md

Described Implementation and Application of the Random Walker

ArkayHub committed:
Update README.md

Minor fixes

Commit:
Changed Documentation

Commit:
Merge remote-tracking branch 'origin/main' into main

Commit:
pylint improvements

BoSott commented on README.md:
```diff
@@ -1,11 +1,34 @@
+- -s : number of steps per walkers
+
+- --save : save figure
 
```
some explanation what each parameter does would be great and what type it requires

BoSott commented on main.py:
```diff
@@ -1,40 +1,55 @@
+        plt.plot(walker.x_positions,
+                 walker.y_positions,
+                 label=str(type(walker).__name__) + ' index: ' + str(walker_index))
+    # optional save the plot
```
the plot is quite messy.. maybe you could add some transparency or just scale it differently or reduce the step size as one cannot identify anything in the plot

juliusernesti replied:
changed the default arguments

BoSott commented on main.py:
```diff
@@ -1,40 +1,55 @@
-# of size equals to the number of size and filled up with 0's
-x = numpy.zeros(n)
-y = numpy.zeros(n)
+def main(steps: int, walkers: int, save: bool, map_seed: int, land_scale: int):
```
i like that saving is made optional

BoSott commented on random_walker.py:
```diff
@@ -0,0 +1,221 @@
+
+    def get_x_factor(self) -> float:
+        """
+        get the x amount of a walk direction
```
minor - capital letter at the beginning of a docstring

BoSott commented on random_walker.py:
```diff
@@ -0,0 +1,221 @@
+
+class WalkDirection(Enum):
+    """Enum Class with all possible walk directions"""
+    Right = 1
```
class constants in UPPER_CASE (at least for pylint)

BoSott commented on random_walker.py:
```diff
@@ -0,0 +1,221 @@
+        """
+        return self.walkDirections[random.randint(0, len(self.walkDirections) - 1)]
+
+    def get_walk_step_length(self, walk_direction: WalkDirection) -> Union[int, float]:
```
unused parameter

BoSott commented on random_walker.py:
```diff
@@ -0,0 +1,221 @@
+
+class WalkDirection(Enum):
+    """Enum Class with all possible walk directions"""
+    Right = 1
+    Left = 2
+    Up = 3
+    Down = 4
+    UpRight = 5
+    UpLeft = 6
+    DownRight = 7
+    DownLeft = 8
```
class constants in UPPER_CASE (at least for pylint)

BoSott commented on random_walker.py:
```diff
@@ -0,0 +1,221 @@
+    return result
+
+
+class Rook(RandomWalker):
```
nice cascading of walker classes and using chess names / characteristics is nice

BoSott commented on playground.py:
```diff
@@ -0,0 +1,90 @@
+
+class Playground:
+    """Defines a playground with obstacles for RandomWalkers"""
+    def __init__(self, scaling: int = 1, x_max: int = 250, y_max: int = 250, seed: int = 0):
```
the fix size of the playground could be changed so that it adapts to the number of walkers -> would make the plot clearer as well

juliusernesti replied:
set default sacling to 4 in the main function

juliusernesti replied:
adapt scaling to number of walkers would not change anything.

BoSott commented on README.md:
```diff
@@ -1,11 +1,34 @@
+- --save : save figure
 
-The Goal is to **practise writing readable, maintainable and reliable code collaboratively.**
 
```
An example how to run it would be good and clean up the description from before :)

BoSott commented on playground.py:
```diff
@@ -0,0 +1,90 @@
+        :return: True for a valid position, else False
+        """
+        position = Point((x_position, y_position))
+        return self.shape.contains(position)
```
this does not work for walkers that "jump" over the obstacle -> one would need to check the area in between as well -> maybe create a line from the current point to the next point and intersect that?
How slow / fast is this contains method?

juliusernesti replied:
That is a simplification of the problem for faster simulation

BoSott commented on playground.py:
```diff
@@ -0,0 +1,90 @@
+    def get_line_segments(self) -> list:
+        """
+        Get all parts from the playground in order to print them with py plot
+
```
don't know where exactly you plot it but a filling the obstacle for clear visual segmentation compared to the rest would be great

juliusernesti replied:
Nice to have but then it will be much more complex.

BoSott commented on README.md:
```diff
@@ -1,11 +1,34 @@
+
+When running our Random-Walker-Application, a number of randomly chosen walkers will walk a set number of steps following their spefific rules. The resulting paths are drawn and returned as a plot.
+For the execution of main.py you can set the following flags:
+- -h : help
+- -w : number of walkers
+
+- -ls: adjust landscale playground scale
+- -ps : set playground/map generation seed
+- -s : number of steps per walkers
+
+- --save : save figure
```
some explanation what each parameter does would be great and what type it requires
bit more descriptive in general
also: some sort to decide which type of walker is chosen would be good

juliusernesti replied:
added option to choose from the walker list

BoSott commented on README.md:
```diff
@@ -1,11 +1,34 @@
 
-The Goal is to **practise writing readable, maintainable and reliable code collaboratively.**
 
 ## Group Exercise
```
clean up the rest of the ReadME

ArkayHub replied:
Done

BoSott commented on main.py:
```diff
@@ -1,40 +1,55 @@
+                 walker.y_positions,
+                 label=str(type(walker).__name__) + ' index: ' + str(walker_index))
+    # optional save the plot
+    if save:
```
![image](https://user-images.githubusercontent.com/73953672/128150847-df291931-552a-44ac-a9b6-39211348a9b7.png)
python mian.py -w 3 -ls 200 -ps 2 -s 100
that is not how it should look like I guess? or is it?
the landscape scaling seems not to be taken into account when you plot the graph

BoSott commented on playground.py:
```diff
@@ -0,0 +1,90 @@
+            holes.append(outer_lake_border)
+            holes.append(inner_lake_border)
+        self.holes = holes
+        self.shape = Polygon(border_polygon, holes)
```
the obstacle implementation is quite neat

BoSott commented on main.py:
```diff
@@ -1,40 +1,55 @@
 #!/usr/bin/env python
```
we are desperately looking for some tests but cannot find any, do you know where they are?:D

BoSott replied:
we also did not write any tests ourselves yet - how would you go about it? comparing random walker to random walker does not really make any sense.. 
maybe a check if a point / line intersects an obstacle would make sense but how would you test the other parts? test the functions separately?

BoSott commented on .gitignore:
```diff
@@ -83,7 +83,7 @@ ipython_config.py
 # pyenv
 .python-version
-
+.idea
```
you might also wanna include the resulting png's in the gitignore to limit the upload to git

Commit:
improoved code quality,
added --names option to specify the walkers to choose from
changed default parameters
added pytest for walking directions

ArkayHub committed:
Update README.md

Added description for default execution without setting any of the provided flags. Cleaned up the initial project part of the README.md

ArkayHub committed:
Create task_specification.md

Added a markdown-file for the initial task specifications.

ArkayHub committed:
Update README.md

Added additional information for playground seeds

Commit:
updated doku

Commit:
updated help

Commit:
Added test_playground.py with testcases for the standard constructor and the class method is_position_in_playground.

Pre-release Adjustments:
- main.py: Only one additional non-default playground is accessible at launch. Accordingly edited the parser argument description for the playground seeds.
- playground.py: Marked the second playground model as "in development" (might never get finished though)

Commit:
Merge remote-tracking branch 'origin/main' into main

ArkayHub committed:
Update README.md

Removed second playground option.

Commit:
added example figures

ArkayHub committed:
Update README.md

Commit:
bugfix legend

Commit:
Merge branch 'main' of https://github.com/juliusernesti/random_walk into main
